### PR TITLE
clickhouse: use `lib.recursiveUpdate` to deep merge extraConfig

### PR DESCRIPTION
**Pull request by conscious-puppet**
previously, was shallow merge to append extra configs, which would have removed some configs. updated to use `lib.recursiveUpdate` to deep merge extraConfig.

### CODE REVIEW

File: nix/clickhouse/default.nix [hunk 1]
```diff
@@ -75,24 +75,26 @@ in
...
+                  path = "${config.dataDir}/clickhouse";
+                  tmp_path = "${config.dataDir}/clickhouse/tmp";
+                  user_files_path = "${config.dataDir}/clickhouse/user_files";
+                  format_schema_path = "${config.dataDir}/clickhouse/format_schemas";
+                  user_directories = {
+                    users_xml = {
+                      path = "${config.package}/etc/clickhouse-server/users.xml";
+                    };
+                  };
+                }
```
Comment (srid) [inline nix/clickhouse/default.nix:97]: An `internal` and `readOnly` option containing this default value would improve a bit on readability.

Comment (conscious-puppet) [reply nix/clickhouse/default.nix:97]: sure, i'll add `internal` and `readOnly` options for the default config

Comment (conscious-puppet) [reply nix/clickhouse/default.nix]: updated with these changes

Comment (srid) [reply nix/clickhouse/default.nix]: Hmm, sorry I wasn't being clear. I meant *one* option like:

```nix
defaultExtraConfig = lib.mkOption {
  type = types.attrs;
  internal = true;
  readOnly = true;
  default = {
    tcp_port = "${toString config.port}";
    ...
  };
};
```

**Review by srid — APPROVED** [review]
`defaultExtraConfig` would be more clearer, other than that looks goods to me.

**Pull request merged**